### PR TITLE
testing init-environment

### LINKED ISSUE #795: Add tests for init-environment command

### PR DESCRIPTION
**Pull request by ratulm**
Fixes #795

---
This change is [Reviewable](https://reviewable.io/reviews/batfish/batfish/798)

### CODE REVIEW

**Comment by dhalperi** [comment]
Reviewed 3 of 3 files at r1.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[tests/basic/commands, line 51 at r1](https://reviewable.io:443/reviews/batfish/batfish/798#-L0waqR3ruVTWkacyuO1:-L0waqR3ruVTWkacyuO2:b2e6lri) ([raw file](https://github.com/batfish/batfish/blob/355dfa61055a09dc05122e334a6dd3f709f7e923/tests/basic/commands#L51)):*
> ```
> 
> # init-environment
> test tests/basic/init-env.ref init-environment interfaceBlacklist=[{hostname="as2border2",interface="GigabitEthernet0/0"}]
> ```

OK

Also wondering, should we have a test in the failure case?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/batfish/batfish/798)*

File: tests/basic/init-env-bad.ref [hunk 1]
```diff
@@ -0,0 +1,28 @@
+{
+  "answerElements" : [
+    {
+      "class" : "org.batfish.common.BatfishException$BatfishStackTrace",
+      "answer" : [
+        "org.batfish.common.BatfishException: Batfish job failed",
+        "   at org.batfish.main.Driver$1.run(Driver.java:384)",
+        "Caused by: org.batfish.common.BatfishException: Cannot continue: environment 'env_0a30fa3d-4fd0-4ffc-8ffc-c4a2434b793b' is invalid:",
```
Comment (dhalperi) [inline tests/basic/init-env-bad.ref]: darnit, this is probably not going to work for determinism. Is there a way to control the environment name?

**Comment by ratulm** [comment]
Review status: 2 of 4 files reviewed at latest revision, 2 unresolved discussions.

---

*[tests/basic/commands, line 51 at r1](https://reviewable.io:443/reviews/batfish/batfish/798#-L0waqR3ruVTWkacyuO1:-L0wf-SNqnAUX6vT9AYd:b-896fix) ([raw file](https://github.com/batfish/batfish/blob/355dfa61055a09dc05122e334a6dd3f709f7e923/tests/basic/commands#L51)):*
<details><summary><i>Previously, dhalperi (Dan Halperin) wrote…</i></summary><blockquote>

OK

Also wondering, should we have a test in the failure case?
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/batfish/batfish/798)*

**Comment by ratulm** [comment]
Review status: 2 of 4 files reviewed at latest revision, 1 unresolved discussion.

---

*[tests/basic/init-env-bad.ref, line 8 at r2](https://reviewable.io:443/reviews/batfish/batfish/798#-L0wdWvdDeGHti-M0qdR-r2-8:-L0wgj26XHZ6K81mDqIy:b-kjvx2e) ([raw file](https://github.com/batfish/batfish/blob/384d04c0f6c8e44f2393a32738a2d3111e296b12/tests/basic/init-env-bad.ref#L8)):*
<details><summary><i>Previously, dhalperi (Dan Halperin) wrote…</i></summary><blockquote>

darnit, this is probably not going to work for determinism. Is there a way to control the environment name?
</blockquote></details>

yes, committed a change for that. but it seems even the stack trace is problematic. 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/batfish/batfish/798)*

**Comment by dhalperi** [comment]
:lgtm:

---

Reviewed 2 of 2 files at r3.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/batfish/batfish/798#-:-L0zP9MgbEIPf4isLwp5:bnfp4nl)*

**Pull request merged**